### PR TITLE
Replace memalloc by new delete

### PR DESCRIPTION
**Pull request by payetvin**
#1304

### CODE REVIEW

File: src/solver/simulation/sim_alloc_probleme_hebdo.cpp [hunk 1]
```diff
@@ -381,201 +372,197 @@ void SIM_AllocationProblemeHebdo(PROBLEME_HEBDO& problem, uint NombreDePasDeTemp
...
+          = new double[NombreDePasDeTemps];
         problem.ReserveJMoins1[k]->ReserveHoraireJMoins1Ref
-          = (double*)MemAlloc(NombreDePasDeTemps * sizeof(double));
+          = new double[NombreDePasDeTemps];
         problem.ResultatsHoraires[k].ValeursHorairesDeDefaillancePositive
-          = (double*)MemAlloc(NombreDePasDeTemps * sizeof(double));
+          = new double[NombreDePasDeTemps];
         problem.ResultatsHoraires[k].ValeursHorairesDENS
-          = (double*)MemAllocMemset(NombreDePasDeTemps * sizeof(double)); // adq patch
+          = new double[NombreDePasDeTemps](); // adq patch
```
Comment (flomnes) [inline src/solver/simulation/sim_alloc_probleme_hebdo.cpp:443]:
```suggestion
          = new double[NombreDePasDeTemps]; // adq patch
```

Comment (flomnes) [reply src/solver/simulation/sim_alloc_probleme_hebdo.cpp:443]: Same for 3 below

**Review by flomnes — CHANGES_REQUESTED** [review]
Please remove CoutsMarginauxDesContraintesDeReserveParZone and its type definition (unused, not freed => memory leak ?)

File: src/solver/simulation/sim_alloc_probleme_hebdo.cpp (hunk 1 above)
Comment (payetvin) [reply src/solver/simulation/sim_alloc_probleme_hebdo.cpp:443]: Used to init to zero, needed to solve valgrind errors

File: src/solver/simulation/sim_alloc_probleme_hebdo.cpp [hunk 2]
```diff
@@ -381,201 +372,197 @@ void SIM_AllocationProblemeHebdo(PROBLEME_HEBDO& problem, uint NombreDePasDeTemp
...
         problem.CaracteristiquesHydrauliques[k]->WaterLayerValues
-          = (double*)MemAllocMemset(100 * sizeof(double));
+          = new double[100];
         problem.CaracteristiquesHydrauliques[k]->InflowForTimeInterval
-          = (double*)MemAllocMemset(100 * sizeof(double));
+          = new double[100];
 
         problem.CaracteristiquesHydrauliques[k]->MaxEnergiePompageParIntervalleOptimise
-          = (double*)MemAllocMemset(7 * sizeof(double));
+          = new double[7];
```
Comment (JasonMarechal25) [inline src/solver/simulation/sim_alloc_probleme_hebdo.cpp:432]: Can be done in a different issue :
Replace hard coded value by constant with meaningful names

**Review by JasonMarechal25 — APPROVED** [review]

**Review by flomnes — APPROVED** [review]

**Pull request merged**